core/src/main/java/io/questdb/griffin/model/WithClauseModel.java

```diff
@@ -24,36 +24,74 @@
 
 package io.questdb.griffin.model;
 
+import io.questdb.std.LowerCaseCharSequenceObjHashMap;
 import io.questdb.std.Mutable;
+import io.questdb.std.ObjList;
 import io.questdb.std.ObjectFactory;
+import org.jetbrains.annotations.Nullable;
 
 public class WithClauseModel implements Mutable {
 
     public static final ObjectFactory<WithClauseModel> FACTORY = WithClauseModel::new;
     private QueryModel model;
+    private LowerCaseCharSequenceObjHashMap<WithClauseModel> originalWithClauses;
+    private int originalWithClausesSize;
     private int position;
+    private LowerCaseCharSequenceObjHashMap<WithClauseModel> withClauses;
+
+    private boolean withClausesInitialized;
 
     private WithClauseModel() {
     }
 
     @Override
     public void clear() {
-        this.position = 0;
-        this.model = null;
+        position = 0;
+        model = null;
+        originalWithClauses = null;
+        originalWithClausesSize = -1;
+        withClauses = null;
```

Comment:
Don't we need to set `withClausesInitialized` to `false` somewhere here?

Reply:
Good catch ! Fixed in 7d0849b9 .